kyaml/fn/framework/processors.go

```diff
@@ -382,29 +418,86 @@ func TemplatesFromDir(dirs ...pkger.Dir) TemplatesFunc {
 				if !strings.HasSuffix(info.Name(), ".template.yaml") {
 					return nil
 				}
-				name := path.Join(dir, info.Name())
-				f, err := pkger.Open(name)
+				t, err := parseTemplate(path.Join(dir, info.Name()))
 				if err != nil {
 					return err
 				}
-				defer f.Close()
 
-				b, err := ioutil.ReadAll(f)
+				pt = append(pt, t)
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pt, nil
+	}
+}
+
+// SchemaDefinitionsFromFile returns a SchemaDefinitionFunc that will load schemas from the provided files.
+// This is a helper to facilitate providing custom resource schemas to a TemplateProcessor.
+func SchemaDefinitionsFromFile(files ...string) SchemaDefinitionFunc {
+	return func() ([]*spec.Definitions, error) {
+		var defs []*spec.Definitions
+		for _, filename := range files {
+			def, err := readSchemaJSON(filename)
```

Comment:
There are some functions in kyaml/openapi, `openapi.SchemaFromFile` or `openapi.AddSchema`? If the goal is to load all schemas from provided files, is there a reason you can't use `openapi.AddSchema` to add all the definitions here, rather than getting the definitions and making a separate call to `openapi.AddDefinitions` in line 228 above?

Reply:
I couldn't use `openapi.SchemaFromFile` because I don't want to read from the local filesystem. Since we're in the function framework, this code is meant to be executed in a Dockerized context. The current way it facilitates this is the use of pkger in its helpers (these new ones are very similar to existing ones for templates). `openapi.SchemaFromFile` is also looking for a top-level "openapi" key that the file I fetched with `kustomize openapi fetch` doesn't have, and does a yaml->json->struct dance I don't need, so it seemed like it was intended for a somewhat different data source.

I could use `openapi.AddSchema` after reading the bytes though. That one is more of an aesthetic choice: the pluggable helpers can generate the definitions however they'd like, and the framework is responsible for managing the global schema. I do also like the ability to throw an error if the document didn't contain any definitions to add.